tests/functions/test_row_to_names.py

```diff
@@ -19,6 +19,18 @@ def test_row_to_names_delete_this_row(dataframe):
     assert df.iloc[2, 2] == 1
     assert df.iloc[2, 3] == "rabbit"
     assert df.iloc[2, 4] == "Cambridge"
+    assert df.iloc[2].name == 2
+
+
+@pytest.mark.functions
+def test_row_to_names_delete_the_row_without_resetting_index(dataframe):
```

Comment:
Going to suggest a test that, perhaps, more generally tests the exact _property_ we'd like to guarantee.

Because the intent of the function is to delete rows without resetting the index, could we do something akin to:

```python
def test_row_to_names_delete_the_row_without_resetting_index(dataframe):
    """Test that executing row_to_names does not reset the index."""
    expected_index = pd.Index(...)
    pd.testing.assert_index_equal(df.index, expected_index)
```

The idea here is that as far as possible, we test desired properties of the resultant dataframe, rather than only test exact values. What we gain here is generality; the test becomes less specific to a particular dataframe's values and much more general. We get a more powerful test. In the case of this function, we want to test that the dataframe index has some form that relates to the original index in a particular way (i.e. it is missing a specified row). 

I'll be the first to admit that the tests that you encountered here as a pattern could have been done better, perhaps. Hopefully, the code sketch above gives you the right ideas, @fireddd. I'd only ask that you do the style of a test I suggested for the two tests that you've added, don't worry about the previously-existing tests.

Could we also make sure there are docstrings for the tests? The intent of the test should be described; I provided an example up there in the code sketch.

Reply:
sure @ericmjl I have change the earlier ones also if you want. That shouldn't be an issue or maybe we can raise another issue for it. I will add the docstrings as well. Thanks for  the suggestion

Reply:
> I can change the earlier ones also if you want. That shouldn't be an issue or maybe we can raise another issue for it.

@fireddd no pressure! Let's handle it this way.

1. If you're up to it, please feel free to change the other tests!
2. Otherwise, the only ones I'll ask you to change are the ones you've added :smile:. 
    1. Definitely raise an issue for this matter. Be sure to reference one of our comments directly so that we can see context easily!